src/components/Table/hoc/withTableSettings/TableColumnSetup/TableColumnSetup.tsx

```diff
@@ -1,270 +1,154 @@
-                        <Icon data={Lock} />
-                    </div>
-                ) : (
-                    <div className={b('tick-wrap', {visible: item.selected})}>
-                        <Icon data={Check} className={b('tick')} width={10} height={10} />
-                    </div>
-                )}
-                <div className={b('title')}>{getItemTitle(item)}</div>
-            </div>
+            <React.Fragment>
+                {renderList()}
+                <Button view="action" className={applyButtonCn} onClick={onApply}>
+                    {i18n('button_apply')}
+                </Button>
+            </React.Fragment>
         );
     };
 
-    const renderStatus = () => {
-        if (!showStatus) {
-            return null;
-        }
-
-        const selected = getCountSelected();
-        const all = propsItems.length;
-        const status = `${selected}/${all}`;
+    const renderControl: TreeSelectProps<unknown>['renderControl'] = ({toggleOpen}) => {
+        const onKeyDown = createOnKeyDownHandler(toggleOpen);
 
-        return <span className={b('status')}>{status}</span>;
+        return (
+            renderSwitcher?.({onClick: toggleOpen, onKeyDown}) || (
+                <Button onClick={toggleOpen} onKeyDown={onKeyDown}>
+                    <Icon data={Gear} />
+                    {i18n('button_switcher')}
+                </Button>
+            )
+        );
     };
 
-    const renderRequiredColumns = () => {
-        const hasRequiredColumns = requiredItems.length;
+    const onOpenChange = (open: boolean) => {
+        setOpen(open);
 
-        if (!hasRequiredColumns) {
-            return null;
+        if (open === false) {
+            const initialItems = prepareItems(propsItems);
+            setItems(initialItems);
         }
-
-        return (
-            <List
-                items={requiredItems}
-                itemHeight={LIST_ITEM_HEIGHT}
-                itemsHeight={getRequiredListHeight}
-                filterable={filterable}
-                renderItem={renderItem}
-                itemsClassName={b('items')}
-                itemClassName={b('item')}
-                virtualized={false}
-            />
-        );
     };
 
-    const renderConfigurableColumns = () => {
-        return (
-            <List
-                items={currentItems}
-                itemHeight={LIST_ITEM_HEIGHT}
-                itemsHeight={getListHeight}
-                sortable={sortable}
-                filterable={filterable}
-                sortHandleAlign={'right'}
-                onSortEnd={makeOnSortEnd(currentItems)}
-                onItemClick={handleItemClick}
-                renderItem={renderItem}
-                itemsClassName={b('items')}
-                itemClassName={b('item')}
-                virtualized={false}
-            />
-        );
-    };
+    const onUpdate = React.useCallback((selectedItemsIds: string[]) => {
```

Comment:
Do we really need memoization here?

Reply:
why not ? (of course, if there is no agreement inside uikit - is there ?)

`deps` array is empty and this property will be passed to `TreeSelect` (so I do not know what `TreeSelect` will really do with this property)

Reply:
We use memoization when it is really needed, not everywhere.
`DndTreeSelect` is not wrapped in `React.memo` so it will rerender every time it's parent rerenders. It seems there's no need in memoization.